changelogs/fragments/async_really_true.yml

```
@@ -0,0 +1,6 @@
+bugfixes:
+- async_wrapper will now use actual booleans vs underlying int 1/0
+- async_status will now use actual booleans vs underlying int 1/0
+- finished test will now use actual booleans vs underlying int 1/0
+- started tests will now use actual booleans vs underlying int 1/0
+- updated powershell side to match
```

These are sorted alphabetically by individual item and intermixed with all other `bugfixes` items in the generated documentation blobs, so unqualified items like `updated powershell side to match` will be nonsensical in the released documentation.

```suggestion
- async tasks - Results from ``async`` module invocations and the ``async_status`` action always return boolean values for ``started`` and ``finished``. Previously, integer and boolean values were mixed.
- Jinja ``started`` and ``finished`` test plugins - Test plugin always returns a boolean value.
```

---

the test plugins always return boolean, that is a nature of being test plugins, that has not changed. What changed is they expect boolean as input. I created an entry per module/action/test as it makes it independently searchable

---

Separate entries or not, you're missing the bigger point: when rendered in the docs as you have them, they're partially/completely nonsensical as they won't be grouped together.

---

and as you have them written, the test plugins do *not* always return booleans, but they need to (hence the 3rd round of review comments to that effect)

---

Individual test plugins must return a bool value to avoid a deprecation warning (and eventual error). For example:

```yaml
- hosts: localhost
  gather_facts: no
  tasks:
    - debug:
        msg: "{{ something is started }}"
      vars:
        something:
          started: not_a_bool
```

This will yield a deprecation warning as it is currently implemented in this PR:

```
[DEPRECATION WARNING]: The test plugin 'ansible.builtin.started' returned a non-boolean result of type <class 'ansible.module_utils._internal._datatag._AnsibleTaggedStr'>. Test plugins must have a boolean result. This feature will be removed from ansible-core version 2.23.  
```

I thought we'd fixed all the core tests in the DT PR, but we didn't. https://github.com/ansible/ansible/pull/85149 fixes all of them.